config/src/config.rs

```diff
@@ -119,34 +136,52 @@ fn check_api_secret_file(
 pub fn initial_setup_wallet(
 	chain_type: &global::ChainTypes,
 	data_path: Option<PathBuf>,
+	create_path: bool,
 ) -> Result<GlobalWalletConfig, ConfigError> {
-	check_api_secret_file(chain_type, data_path.clone(), OWNER_API_SECRET_FILE_NAME)?;
-	check_api_secret_file(chain_type, data_path.clone(), API_SECRET_FILE_NAME)?;
+	if create_path {
+		if let Some(p) = data_path.clone() {
+			fs::create_dir_all(p)?;
+		}
+	}
 	// Use config file if current directory if it exists, .grin home otherwise
-	if let Some(p) = check_config_current_dir(WALLET_CONFIG_FILE_NAME) {
-		GlobalWalletConfig::new(p.to_str().unwrap())
+	let (path, config) = if let Some(p) = check_config_current_dir(WALLET_CONFIG_FILE_NAME) {
+		let mut path = p.clone();
+		path.pop();
+		(path, GlobalWalletConfig::new(p.to_str().unwrap())?)
 	} else {
 		// Check if grin dir exists
-		let grin_path = match data_path {
+		let grin_path = match data_path.clone() {
 			Some(p) => p,
-			None => get_grin_path(chain_type)?,
+			None => get_grin_path(chain_type, create_path)?,
 		};
 
 		// Get path to default config file
 		let mut config_path = grin_path.clone();
 		config_path.push(WALLET_CONFIG_FILE_NAME);
 
 		// Return defaults if file doesn't exist
-		if !config_path.exists() {
-			let mut default_config = GlobalWalletConfig::for_chain(chain_type);
-			default_config.config_file_path = Some(config_path);
-			// update paths relative to current dir
-			default_config.update_paths(&grin_path);
-			Ok(default_config)
-		} else {
-			GlobalWalletConfig::new(config_path.to_str().unwrap())
+		match config_path.clone().exists() {
+			false => {
+				let mut default_config = GlobalWalletConfig::for_chain(chain_type);
+				default_config.config_file_path = Some(config_path);
+				// update paths relative to current dir
+				default_config.update_paths(&grin_path);
+				(grin_path, default_config)
+			}
+			true => {
+				let mut path = config_path.clone();
+				path.pop();
+				(
+					path,
+					GlobalWalletConfig::new(config_path.to_str().unwrap())?,
+				)
+			}
```

Comment:
Considering using `if/else` instead of `match` here
```
		// Return defaults if file doesn't exist
		if config_path.clone().exists() {
			let mut default_config = GlobalWalletConfig::for_chain(chain_type);
			default_config.config_file_path = Some(config_path);
			// update paths relative to current dir
			default_config.update_paths(&grin_path);
			(grin_path, default_config)
		} else {
			let mut path = config_path.clone();
			path.pop();
			(
				path,
				GlobalWalletConfig::new(config_path.to_str().unwrap())?,
			)
		}
```

Reply:
I kind of prefer match syntax even for simple cases like this, it makes the false and true conditions more explicit, and harder to mix them up when maintaining (like you may have done in your example) (:rofl: Sorry!)

Reply:
I understand 😆